src/app/shared/stack-blitz/stack-blitz-writer.ts

```diff
@@ -140,26 +143,30 @@ export class StackBlitzWriter {
       'dependencies',
       JSON.stringify(isTest ? testDependencies : dependencies));
 
-    return new Promise(resolve => {
-      const templateContents = (isTest ? TEST_TEMPLATE_FILES : TEMPLATE_FILES)
-        .map(file => this._readFile(form,
-          data,
-          file,
-          isTest ? TEST_TEMPLATE_PATH : TEMPLATE_PATH,
-          isTest));
+    // Run outside the zone since this form doesn't interact with Angular
+    // and the file requests can cause excessive change detections.
+    await this._ngZone.runOutsideAngular(() => {
+      const fileReadPromises: Promise<void>[] = [];
+
+      // Read all of the template files.
+      (isTest ? TEST_TEMPLATE_FILES : TEMPLATE_FILES).forEach(file => fileReadPromises.push(
+        this._loadAndAppendFile(form, data, file, isTest ? TEST_TEMPLATE_PATH : TEMPLATE_PATH,
+          isTest)));
 
-      const exampleContents = data.exampleFiles
-        .map(file => this._readFile(form, data, file, baseExamplePath, isTest));
+      // Read the example-specific files.
+      data.exampleFiles.forEach(file => fileReadPromises.push(this._loadAndAppendFile(form, data,
+        file, baseExamplePath, isTest)));
 
       // TODO(josephperrott): Prevent including assets to be manually checked.
       if (data.selectorName === 'icon-svg-example') {
-        this._readFile(form, data, 'assets/img/examples/thumbup-icon.svg', '', isTest, false);
+        fileReadPromises.push(this._loadAndAppendFile(form, data,
+          'assets/img/examples/thumbup-icon.svg', '', isTest, false));
       }
 
-      Promise.all(templateContents.concat(exampleContents)).then(() => {
```

Comment:
Fun fact: this `Promise.all` didn't actually do anything, because the passed-in array was filled with `undefined` values.

Reply:
😬